sdv/data_processing/data_processor.py

```diff
@@ -417,8 +417,13 @@ def transform(self, data, is_condition=False):
         if not self.fitted:
             raise NotFittedError()
 
+        # Filter columns that can be transformed
+        columns = [
+            column for column in self.get_sdtypes(primary_keys=not is_condition)
+            if column in data.columns
+        ]
         LOGGER.debug(f'Transforming constraints for table {self.table_name}')
-        data = self._transform_constraints(data, is_condition)
+        data = self._transform_constraints(data[columns], is_condition)
```

Comment:
This is because the `CND_IDX` that we create, for some reason this is not ignoring it. This was being done in the legacy code https://github.com/sdv-dev/SDV/blob/26ec69f5bbe5102e68f47811d663012e295b3880/sdv/metadata/table.py#L654 but since we removed `anonymization` we skipped it.
The only part that is strange to me is that if we don't pass the `primary_keys` here when it's not a condition it fails.

Reply:
That is strange. We might want to investigate that a bit